clang/lib/Sema/SemaChecking.cpp

```diff
@@ -8198,20 +8198,48 @@ static bool IsStdFunction(const FunctionDecl *FDecl,
   return true;
 }
 
+enum class MathCheck { NaN, Inf };
+static bool IsSpecialMathFunction(StringRef calleeName, MathCheck Check) {
```

Comment:
Err "special math function" may be a loaded term: https://eel.is/c++draft/numerics#sf.cmath

How about `IsInfOrNanFunction()` instead?